### PR TITLE
Use pytest fixtures in unit tests instead of `from_name` methods

### PR DESCRIPTION
**Pull request by maxnoe**
We are preparing a number of changes to the data model, including also the instrument model and dropping support for older versions.

This is very hard to do when the tests are relying on very old files on the data server accessed via `from_name`.

Here I replace all use in our unit tests of `from_name` with pytest fixtures, which are defined mostly in `ctapipe/conftest` so we can change them in a single location.

Also, most fixtures are updated to use prod5 data.

### CODE REVIEW

**Comment by kosack** [comment]
what's the reasoning for this?

**Comment by maxnoe** [comment]
@kosack As explained in #1978, I think these methods do more harm than good. 

They make it very easy to make mistakes like loading the wrong instrument compared to the data being analyzed. I stopped counting the occurences I had to change code from doing `from_name` to reading the subarray from the input file / just using the one provided by the event source.

**Comment by maxnoe** [comment]
I think the use case for our unit tests is much better solved by defining common fixtures. 

It reduced the code already by quite a lot. I could also imagine just doing the changes in our tests and keeping `from_name` for now, but the main thing is that it was abused far too often resulting in many errors.

**Comment by maxnoe** [comment]
@kosack I removed the "remove from_name" part for now and only included the changes to the fixtures.

It would make it much easier to test the data model changes without constantly having to update files on the data server to include this.

**Review by kosack — APPROVED** [review]

**Comment by kosack** [comment]
This is nice in that we don't rely on the data format.  However, what about tests to read old data formats?  We should still have those (perhaps not now but after the 1.0 release,  to ensure backward compatibility)

**Comment by maxnoe** [comment]
The point is, that we now do want do break backwards compatibilty.

For me this means that tests checking compatibility to old data should be added after we merge the current PRs changing the data model, make a release and first thing we add tests testing files pepduced with that release?

**Comment by maxnoe** [comment]
I would propose the following:

* Merge the remaining, data model changing PRs
* Make the 0.16.0 release
* Process test data files using the released version and add them to the data server in some meaningful structure
* Add (parametrized) tests checking we are still compatible with all data model versions we want to support (i.e. for now the current one for 0.16.0 and the currently implemented on, e.g.:

``
@pytest.fixture("session", params=["v0.16.0", "current"])
def prod5_dl1_files(request, prod5_dl1_file):
    if request.param == "current":
        return prod5_dl1_file
    return get_dataset_path(f"prod5/{request.param}/test.dl1.h5")
```

File: ctapipe/image/tests/test_hillas.py [hunk 1]
```diff
@@ -259,18 +257,18 @@ def test_single_pixel():
...
     """
     np.random.seed(42)
 
     telescope_frame = TelescopeFrame()
-    camera_frame = CameraFrame(focal_length=28 * u.m)
 
-    geom = CameraGeometry.from_name("LSTCam")
-    geom.frame = camera_frame
+    geom = prod5_lst.camera.geometry
+    telescope_frame = TelescopeFrame()
```
Comment (nbiederbeck) [inline ctapipe/image/tests/test_hillas.py:270]: duplicated line

Comment (maxnoe) [reply ctapipe/image/tests/test_hillas.py:268]: fixed

**Comment by maxnoe** [comment]
@kosack @nbiederbeck @LukasNickel Please review again

**Review by kosack — APPROVED** [review]

**Pull request merged**